expression/builtin_time.go

```diff
@@ -2660,7 +2660,11 @@ func (du *baseDateArithmitical) getIntervalFromDecimal(ctx sessionctx.Context, a
 	if isNull || err != nil {
 		return "", true, err
 	}
-
+	neg := false
+	if interval != "" && interval[0] == '-' {
```

Comment:
what if the interval string is `--12`? should it be regarded as `+`?

Reply:
The interval string is converted from Decimal.  `--12` is an invalid Decimal, and it will convert to `12`